stl/inc/expected

```diff
@@ -308,7 +308,11 @@ public:
         : _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...), _Has_value(false) {}
 
     // [expected.object.dtor]
-    constexpr ~expected() noexcept {
+    constexpr ~expected()
+#ifndef __clang__ // TRANSITION, LLVM-59854
+        noexcept
+#endif // __clang__

```

Comment:
Nitpick, no change requested: Ideally I'd like to see `^^^ no workaround ^^^` here and below, to help us remember what to do later. This isn't worth resetting testing, though.

Reply:
Fix pending on Casey's miscellaneous branch.